### DISCUSSION TITLE
Does this tool support Transitive dependency & License?

### DISCUSSION THREAD

**Discussion by rohitcoder** [CycloneDX/cyclonedx-python#485] [Q&A] [answered]
Hi All,

I was checking this tool, i was trying to generate SBOM of my requirements.txt file and it's licenses. I ran below command
``cyclonedx-py -r -i dvpwa/requirements.txt``
but, this doesn't list any transitive dependencies or it's license, i remember i used cyclonedx for some java projects, and it was listing transitive deps with license, is it possible for python requirements.txt / piplock ?

**Comment by jkowalleck** [comment, marked as answer]
Transitive dependencies are not fetched from requirements files like you are providing it.

Please provide a full set of dependencies as generated via `pip-compile` from [`pip-tools`](https://pypi.org/project/pip-tools/).
Or provide a full list of your installed python packages/modules from `pip freeze` - as described here: <https://cyclonedx-bom-tool.readthedocs.io/en/latest/usage.html?highlight=Requirements#requirements>
